server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2349,7 +2350,7 @@ public void acquireReplicaOperationPermit(final long opPrimaryTerm, final long g
                                 logger.info("detected new primary with primary term [{}], global checkpoint [{}], max_seq_no [{}]",
                                              opPrimaryTerm, currentGlobalCheckpoint, maxSeqNo);
                                 if (currentGlobalCheckpoint < maxSeqNo) {
-                                    resetEngineToGlobalCheckpoint();
+                                    resetEngineToGlobalCheckpoint(maxSeqNoOfUpdatesOrDeletes);
```

Comment:
why do we need to set the maxSeqNoUpdatesOrDeletes here? can't we let it be what engine does it self? I think it's also risky because we recovery from the translog (and need make the optimization doesn't create duplicates in thas process)

Reply:
Yes, we can let an engine take care itself. The reason I use the max_seq_no_of_updates from the primary, so at the end of a test, all replicas and its primary have the same max_seq_no_of_updates. I will update this.